### PR TITLE
feat(testutils): Show some useful warnings

### PR DESCRIPTION
**Pull request by doniyor2109**

### CODE REVIEW

**Comment by doniyor2109** [comment]
I am not sure about the warning messages. 
Here some examples:
`Unmatched GET http://host/bar`

`Warning: foo is mocked but not called`

@umidbekkarimov Do we need more information here?

File: packages/testutils/src/__tests__/mockEndpoint.spec.ts [hunk 1]
```diff
@@ -368,3 +368,50 @@ it('accepts request instance', async () => {
...
 });
+
+describe('warning unmocked endpoint', () => {
+  beforeEach(() => {
+    jest.spyOn(console, 'warn').mockImplementation();
+  });
+
+  it('calls fetch without mocks', async () => {
+    try {
+      await fetch('http://host/bar');
```
Comment (umidbekkarimov) [inline packages/testutils/src/__tests__/mockEndpoint.spec.ts]: use `.rejects` here https://jestjs.io/docs/en/tutorial-async#rejects

File: packages/testutils/src/__tests__/mockEndpoint.spec.ts [hunk 2]
```diff
@@ -368,3 +368,50 @@ it('accepts request instance', async () => {
     new Response(null, { status: 500 }),
   );
 });
+
+describe('warning unmocked endpoint', () => {
```
Comment (umidbekkarimov) [inline packages/testutils/src/__tests__/mockEndpoint.spec.ts]: we don't need `describe` here, `it('warns about not matched endpoints')` will be enough.

File: packages/testutils/src/__tests__/mockEndpoint.spec.ts [hunk 3]
```diff
@@ -368,3 +368,50 @@ it('accepts request instance', async () => {
...
+  beforeEach(() => {
+    jest.spyOn(console, 'warn').mockImplementation();
```
Comment (umidbekkarimov) [inline packages/testutils/src/__tests__/mockEndpoint.spec.ts]: you have only one test here, you can mock it inside the test

File: packages/testutils/src/__tests__/mockEndpoint.spec.ts [hunk 4]
```diff
@@ -368,3 +368,50 @@ it('accepts request instance', async () => {
...
+    `);
+  });
+});
+
+describe('warns mocked but not called', () => {
+  beforeEach(() => {
+    jest.spyOn(console, 'warn').mockImplementation();
+  });
+
+  afterAll(() => {
```
Comment (umidbekkarimov) [inline packages/testutils/src/__tests__/mockEndpoint.spec.ts]: can you merge `describe`, `beforeEach` and `afterAll` with `it`.

File: packages/testutils/src/__tests__/mockEndpoint.spec.ts [hunk 5]
```diff
@@ -368,3 +368,50 @@ it('accepts request instance', async () => {
...
+      await fetch('http://host/bar');
+    } catch (e) {
+      // ignore error
+    }
+    // eslint-disable-next-line no-console
+    expect(console.warn).toHaveBeenCalledTimes(1);
+    // eslint-disable-next-line no-console
+    expect(console.warn).toHaveBeenLastCalledWithMatchingInlineSnapshot(`
+      Array [
+        "Unmatched GET http://host/bar",
```
Comment (umidbekkarimov) [inline packages/testutils/src/__tests__/mockEndpoint.spec.ts]: maybe something like `Unmatched 'GET' request to 'http://host/bar'`

File: setupTests.ts [hunk 6]
```diff
@@ -12,13 +12,11 @@ declare global {
 }
 
 beforeEach(() => {
-  jest.spyOn(console, 'warn');
+  jest.spyOn(console, 'warn').mockImplementation();
```
Comment (umidbekkarimov) [inline setupTests.ts]: why do you mock that?

File: setupTests.ts [hunk 7]
```diff
@@ -12,13 +12,11 @@ declare global {
...
   jest.spyOn(console, 'error');
 });
 
 afterEach(() => {
-  // eslint-disable-next-line no-console
-  expect(console.warn).not.toBeCalled();
```
Comment (umidbekkarimov) [inline setupTests.ts]: actually, we need that

**Pull request merged**